### ISSUE TITLE
`WebSocket` (client) hang on repeated large buffer sends

### ISSUE THREAD

**Issue by bwasti** [oven-sh/bun#1344]
### Version

0.2.1

### Platform

Darwin bwasti-mbp 21.6.0 Darwin Kernel Version 21.6.0: Mon Aug 22 20:19:52 PDT 2022; root:xnu-8020.140.49~2/RELEASE_ARM64_T6000 arm64

### What steps will reproduce the bug?

Using the README echo server:

```javascript
Bun.serve({
  websocket: {
    message(ws, message) {
      ws.send(message);
    },
  },

  fetch(req, server) {
    if (server.upgrade(req)) {
      return;
    }

    return new Response("Regular HTTP response");
  },
});
```

and client:

```javascript
const socket = new WebSocket('ws://0.0.0.0:3000')

let i = 0
socket.addEventListener('message', (event) => {
    const t = new Float32Array(event.data)
    console.log('Message from server ', t.length, i++);
    if (i === 1000) {
      socket.close()
    }
});

socket.addEventListener('open', (event) => {
    const b = (new Float32Array(128 * 128)).buffer
    for (let i = 0; i < 1000; ++i) {
      socket.send(b)
    }
});
```

### How often does it reproduce? Is there a required condition?

Every time, the program hangs at around 800 executions on my M1.

### What is the expected behavior?

Shouldn't hang

### What do you see instead?

<img width="239" alt="Screen_Shot_2022-10-18_at_11 25 23_AM" src="https://user-images.githubusercontent.com/4842908/196483318-419d4cfb-727a-4f79-8d60-c7399aea547f.png">

stuck here

**Comment by Kapsonfire-DE** [comment]
Seems to websocket client related. browser works fine

**Comment by Jarred-Sumner** [comment]
snippet that determines this is a client-side issue and not a server issue:
```js
Bun.serve({
  websocket: {
    message(ws, message) {
      ws.send(message);
    },
  },

  fetch(req, server) {
    if (server.upgrade(req)) {
      return;
    }

    return new Response(
      `<html>
      <head>
        <script>
        const socket = new WebSocket('ws://0.0.0.0:3000')
        socket.binaryType = 'arraybuffer'

        let i = 0
        socket.addEventListener('message', (event) => {
            const t = new Float32Array(event.data)
            console.log('Message from server ', t.length, i++);
            if (i === 1000) {
              socket.close()
            }
        });
        
        socket.addEventListener('open', (event) => {
            const b = (new Float32Array(128 * 128)).buffer
            for (let i = 0; i < 1000; ++i) {
              socket.send(b)
            }
        });
        </script>
      </head>
      <body>
        Open the console
        </body>
    </html>`,
      {
        headers: {
          "Content-Type": "text/html; charset=utf-8",
        },
      }
    );
  },
});

```

**Comment by Kapsonfire** [comment]
@Jarred-Sumner 
you need to replace 0.0.0.0:3000 to 127.0.0.1:3000
0.0.0.0 is no valid address...
and why the heck does it work in bun?

**Comment by Jarred-Sumner** [comment]
`0.0.0.0` is a valid address, it means "anything ipv4". It is similar to "::1" 

But that bun doesn't seem to work with `localhost` is a bug and that "0.0.0.0" is what is returned is stupid

<img width="568" alt="image" src="https://user-images.githubusercontent.com/709451/196623318-d213c09a-c08f-41f6-b850-8c4ff00168e5.png">

**Comment by Kapsonfire** [comment]
0.0.0.0 throw errors in browsers - its not valid
0.0.0.0 is per standard not routable

![image](https://user-images.githubusercontent.com/67053124/196624241-61fc2661-b0f4-4f1d-a76b-123d2f68b69c.png)

**Comment by Kapsonfire** [comment]
Linux/OSx seems to routing 0.0.0.0 to 127.0.0.1 - but its invalid per rfc and for windows
![image](https://user-images.githubusercontent.com/67053124/196624886-4e272578-cc04-460b-a2e8-202dd4c673ab.png)

![image](https://user-images.githubusercontent.com/67053124/196624997-19c41625-fb31-460b-b3a0-2b0d955ab683.png)

**Comment by Jarred-Sumner** [comment]
This is fixed in v0.5.2 thanks to @cirospaciari
